cli/defenseclaw/commands/cmd_upgrade.py

```
@@ -1822,6 +1854,345 @@ def _fetch_latest_version() -> str | None:
+    except (OSError, subprocess.SubprocessError) as exc:
+        ux.err(f"Release-owned resolver authentication failed: {exc}", indent="  ")
+        ux.subhead(
+            "No configuration, service, migration cursor, or installed artifact was changed.",
+            indent="    ",
+        )
+        raise SystemExit(1) from exc
+    raise SystemExit(completed.returncode)
+
+
+@contextmanager
+def _authenticated_release_resolver(version: str):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `_authenticated_release_resolver`

(ANN202)